### ISSUE TITLE
jq parsing errors

### ISSUE THREAD

**Issue by gregorkoerdt** [Iserv-Group/Zabbix-Meraki#4]
I get parsing errors on each check. The autodiscovery is working. It created the hosts and also all status and wan checks. Using Debian 10 with Zabbix 5.2 also tested in 5.4, same errors.

Preprocessing failed for: jq - commandline JSON processor [version 1.5-1-a5b5cbe].Usage: jq [options] <jq filter> [file...]...
1. Failed: cannot extract value from json by path "$[?(@.serial == 'XXX-XXXX-XXXX')].status": cannot parse as a valid JSON object: invalid object format, expected opening character '{' or '[' at: 'jq - commandline JSON processor [version 1.5-1-a5b5cbe]
Usage: jq [options] <jq filter> [file...]
jq is a tool for processing JSON inputs, applying t

Preprocessing failed for: jq - commandline JSON processor [version 1.5-1-a5b5cbe].Usage: jq [options] <jq filter> [file...]...
1. Failed: cannot extract value from json by path "$[?(@.serial == 'XXX-XXXX-XXXX')].uplinks": cannot parse as a valid JSON object: invalid object format, expected opening character '{' or '[' at: 'jq - commandline JSON processor [version 1.5-1-a5b5cbe]
Usage: jq [options] <jq filter> [file...]
jq is a tool for processing JSON inputs, applying t

Preprocessing failed for: jq - commandline JSON processor [version 1.5-1-a5b5cbe].Usage: jq [options] <jq filter> [file...]...
1. Failed: cannot extract value from json by path "$.*.status": cannot parse as a valid JSON object: invalid object format, expected opening character '{' or '[' at: 'jq - commandline JSON processor [version 1.5-1-a5b5cbe]
Usage: jq [options] <jq filter> [file...]
jq is a tool for processing JSON inputs, applying t

**Comment by zdegraaf** [comment]
Does the data pull item have any output? Do you have any information in your meraki.json file? That looks like an error that you'd see if there was nothing being retrieved by the data pull.

**Comment by mark-cobra** [comment]
I think I got the same error
![image](https://user-images.githubusercontent.com/58164816/126459245-65371ed9-530c-4ad6-b9a0-e73a279b6006.png)
I have information in my meraki.json file that is generated by meraki_status.sh

**Comment by franko108** [comment]
I have the same problem as both above. The zabbix version is 5.4.  
Host discovery **Meraki Host Creation** doing a nice job and creating the hosts, even creating   
On _Discovery rules_, there is error:
```
Discover Meraki Devices  Invalid discovery rule value: cannot parse as a valid JSON object: invalid JSON object value starting character at: ']}'
Discovered WAN Connections  Invalid discovery rule value: cannot parse as a valid JSON object: invalid JSON object value starting character at: ']}'
```
My jsons are on the same path as the scripts - as defined by default in your scripts. 
_meraki.json_ and _meraki.networks.json_ are created and existing in the same directory. The json format looks valid.  
Zabbix is the owner of the files.

**Comment by zdegraaf** [comment]
@franko108 @mark-cobra 

Please open the failing discovery item, select the "Test" option, then "Get value". 
![image](https://user-images.githubusercontent.com/69164506/126678659-58f6a45e-cf12-464b-b6b2-8f2df8f9a536.png)

Click the edit button to get the full output, it should look something like the following
`{"data":[{   "{#NAME}": "Some Network Name",   "{#INT_IP}": "55.55.55.55",   "{#INT_NAME}": "wan1",   "{#NOTES}": "Some Network Notes",   "{#NET_NAME}": "N_666666666666666666",   "{#SERIAL}": "AB12-CD34-FG56" }]}`

If it looks similar to the above, run it through a json validator like the following site to see if values being pulled in a specific are breaking the JSON format.
https://jsonlint.com/

**Comment by mark-cobra** [comment]
![image](https://user-images.githubusercontent.com/58164816/127178122-d7429342-7355-4cf5-9798-78fd7a358a11.png)

The meraki.json file on the server has data.

I'm confused, what do I need to fill in for the interface for the host. I'm expecting the domain name of Meraki.

**Comment by zdegraaf** [comment]
> ![image](https://user-images.githubusercontent.com/58164816/127178122-d7429342-7355-4cf5-9798-78fd7a358a11.png)

In this picture example, it doesn't include the Network ID value, like you are testing from the template or something. Make sure to test from a created Host(that represents a Meraki Network)

**Comment by mark-cobra** [comment]
I have solved the problem, I did not understand this step in the instructions "11. Apply the Meraki Network Discovery template to your Zabbix host, then run the discovery rule." If someone is also confused by this, I try to explain.
We need to add the Network Discovery template to the "Zabbix server" host you probably already have to monitor your zabbix server.
After you execute the discovery rule a new host will be automatically created with the network name and the devices are added.
